lib/hotbar.js

```diff
@@ -20,26 +20,48 @@ class HotBar extends LitElement {
     })
 
     this.bot.inventory.on('updateSlot', (slot, oldItem, newItem) => {
-      if (slot >= this.bot.inventory.hotbarStart + 9) return
-      if (slot < this.bot.inventory.hotbarStart) return
+      if ((slot >= this.bot.inventory.hotbarStart + 9 || slot < this.bot.inventory.hotbarStart) && slot !== this.bot.getEquipmentDestSlot('off-hand')) return
 
       const sprite = newItem ? invsprite[newItem.name] : invsprite.air
-      const slotImage = this.shadowRoot.getElementById('hotbar-' + (slot - this.bot.inventory.hotbarStart))
+      const slotId = slot === this.bot.getEquipmentDestSlot('off-hand') ? 'offhand' : (slot - this.bot.inventory.hotbarStart)
+      const slotImage = this.shadowRoot.getElementById('hotbar-' + slotId)
       slotImage.style['background-position-x'] = `-${sprite.x * 2}px`
       slotImage.style['background-position-y'] = `-${sprite.y * 2}px`
       slotImage.innerHTML = newItem?.count > 1 ? newItem.count : ''
+      if (slot === this.bot.getEquipmentDestSlot('off-hand')) {
+        const offhandImg = this.shadowRoot.getElementById('hotbar-offhand-image')
+        if (newItem) offhandImg.show()
+        else offhandImg.hide()
+      }
     })
   }
 
   async reloadHotbar () {
-    for (let i = 0; i < 9; i++) {
-      const item = this.bot.inventory.slots[this.bot.inventory.hotbarStart + i]
+    const slotsToReload = {
+      0: this.bot.inventory.hotbarStart,
+      1: this.bot.inventory.hotbarStart + 1,
+      2: this.bot.inventory.hotbarStart + 2,
+      3: this.bot.inventory.hotbarStart + 3,
+      4: this.bot.inventory.hotbarStart + 4,
+      5: this.bot.inventory.hotbarStart + 5,
+      6: this.bot.inventory.hotbarStart + 6,
+      7: this.bot.inventory.hotbarStart + 7,
+      8: this.bot.inventory.hotbarStart + 8,
+      offhand: this.bot.getEquipmentDestSlot('off-hand')
+    }
+    Object.keys(slotsToReload).forEach(key => {
+      const item = this.bot.inventory.slots[slotsToReload[key]]
       const sprite = item ? invsprite[item.name] : invsprite.air
-      const slotImage = this.shadowRoot.getElementById('hotbar-' + i)
+      const slotImage = this.shadowRoot.getElementById('hotbar-' + key)
       slotImage.style['background-position-x'] = `-${sprite.x * 2}px`
       slotImage.style['background-position-y'] = `-${sprite.y * 2}px`
       slotImage.innerHTML = item?.count > 1 ? item.count : ''
-    }
+      if (key === 'offhand') {
+        const offhandImg = this.shadowRoot.getElementById('hotbar-offhand-image')
+        if (item) offhandImg.show()
+        else offhandImg.hide()
+      }
+    })
```

Comment:
Why use `Object.keys`?
```suggestion
    function updateSlot (key, item) {
      const sprite = item ? invsprite[item.name] : invsprite.air
      const slotImage = this.shadowRoot.getElementById('hotbar-' + key)
      slotImage.style['background-position-x'] = `-${sprite.x * 2}px`
      slotImage.style['background-position-y'] = `-${sprite.y * 2}px`
      slotImage.innerHTML = item?.count > 1 ? item.count : ''
      if (key === 'offhand') {
        const offhandImg = this.shadowRoot.getElementById('hotbar-offhand-image')
        if (item) offhandImg.show()
        else offhandImg.hide()
      }
    }
    for (let i = 0; i < 9; i++) {
      updateSlot(i, this.bot.inventory.slots[this.bot.inventory.hotbarStart + i])
    }
    updateSlot('offhand', this.bot.getEquipmentDestSlot('off-hand'))
```

Reply:
Yea, this is a better way of doing it.